### PR TITLE
Configuration Management

### PR DESCRIPTION
**Pull request by jackzampolin**
- [x] `lens chains add [chain-name]` - adds configuration from the `chain-registry` to the local config
- [x] `lens chains delete [chain-name]`
- [x] `lens chains edit [key] [value]`
- [x] `lens chains list`
- [x] `lens chains show [chain-name]` - chain name is optional here, will show default if no name passed
- [x] `lens chains set-default [chain-name]`

### CODE REVIEW

File: cmd/chains.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+			fmt.Println(string(bz))
+			return nil
+		},
+	}
+	return cmd
+}
+
+func fetchRegistryChains(ctx context.Context) ([]string, error) {
+	chains := []string{}
+	tree, res, err := github.NewClient(nil).Git.GetTree(ctx, "cosmos", "chain-registry", "master", true)
```
Comment (joeabbey) [inline cmd/chains.go:60]: TODO: Make this a config option?

File: cmd/config.go [hunk 2]
```diff
@@ -114,22 +106,42 @@ func defaultConfig(keyHome string, debug bool) []byte {
...
+				GasAdjustment:  1.2,
+				GasPrices:      "0.01uosmo",
+				KeyDirectory:   keyHome,
+				Debug:          debug,
+				Timeout:        "20s",
+				OutputFormat:   "json",
+				BroadcastMode:  "block",
+				SignModeStr:    "direct",
+				Modules:        modules,
+			},
```
Comment (joeabbey) [inline cmd/config.go:144]: Do we need these any more?

**Review by joeabbey — APPROVED** [review]
One quick comment on the "default" chains

And a TODO: for allowing "override-by-config" for chain registry

File: cmd/config.go (hunk 2 above)
Comment (jackzampolin) [reply cmd/config.go:144]: I like populating a default config with some stuff. Are you opposed to keeping this?

Comment (joeabbey) [reply cmd/config.go:144]: I'd rather it get populated from the registry :) 

```
                        defaults := ["cosmoshub", "osmosis"]
			for _, chain := range defaults {
				ch, err := getChainConfigFromRegistry(chain, path.Join(home, "keys"), debug)
				if err != nil {
					return err
				}
				config.Chains[chain] = ch
			}

```

**Pull request merged**